packages/core/core-flows/src/definition/cart/workflows/add-shipping-method-to-cart.ts

```diff
@@ -78,10 +79,13 @@ export const addShippingMethodToWorkflow = createWorkflow(
     const shippingMethods = addShippingMethodToCartStep({
       shipping_methods: shippingMethodInput,
     })
-    refreshCartPromotionsStep({ id: input.cart_id })
-    updateTaxLinesStep({
-      cart_or_cart_id: input.cart_id,
-      shipping_methods: shippingMethods,
-    })
+
+    parallelize(
+      refreshCartPromotionsStep({ id: input.cart_id }),
+      updateTaxLinesStep({
```

Comment:
One thing to check here is that the adjustments created by promotions would be accounted for in tax lines. Im not sure if this can be paralrllized.

Reply:
Ah ok, might be a missed implementation. Ignore this comment, I'll create a ticket 🤞🏻

Reply:
so that can't be in parallel?
should I merge this PR and later you remove it from the parallelize?

Reply:
But I believe the taxes and adjustments are independent and only calculated when we display totals, isn't it?

Reply:
You're right! Sorry, I got confused between tax rates and tax amount